pkgs/by-name/ev/evcxr-jupyter/package.nix

```
@@ -0,0 +1,67 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  jupyter,
+  cmake,
+  pkg-config,
+  openssl,
+  zlib,
+  zeromq,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "evcxr-jupyter";
+  version = "0.21.1";
+  src = fetchFromGitHub {
+    owner = "evcxr";
+    repo = "evcxr";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-8PjZFWUH76QrA8EI9Cx0sBCzocvSmnp84VD7Nv9QMc8=";
+  };
+  cargoHash = "sha256-hE/O6lHC0o+nrN4vaQ155Nn2gZscpfsZ6o7IDi/IEjI=";
+  doCheck = false;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    jupyter
+  ];
+
+  buildInputs = [
+    openssl
+    zlib
+    zeromq
+  ];
+
+  postInstall = ''
+    # Create the directory structure
+    mkdir -p $out/share/jupyter/kernels/rust
+
+    # Create a kernel.json file manually with the exact binary path
+    cat > $out/share/jupyter/kernels/rust/kernel.json << EOF
+    {
+      "argv": ["$out/bin/evcxr_jupyter", "--control_file", "{connection_file}"],
+      "display_name": "Rust",
+      "language": "rust",
+      "interrupt_mode": "message"
+    }
+    EOF
+
+    # Verify the binary exists and is executable
+    if [ ! -x "$out/bin/evcxr_jupyter" ]; then
+      echo "Error: evcxr_jupyter binary not found or not executable at $out/bin/evcxr_jupyter"
+      ls -la $out/bin/
+      exit 1
+    fi
+
+    echo "evcxr_jupyter binary found at $out/bin/evcxr_jupyter"
+  '';
+
+  meta = {
+    description = "Jupyter kernel for Rust";
+    homepage = "https://github.com/evcxr/evcxr";
+    license = lib.licenses.mit;
```

```suggestion
    license = with lib.licenses; [
      asl20
      mit
    ];
```

https://github.com/evcxr/evcxr?tab=readme-ov-file#license